editor/header/publish-button/label.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { flowRight } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { withAPIData } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import {
+	isCurrentPostPublished,
+	isEditedPostBeingScheduled,
+} from '../../selectors';
+
+export function PublishButtonLabel( {
+	isPublished,
+	isBeingScheduled,
+	user,
+} ) {
+	const isContributor = user.data && ! user.data.capabilities.publish_posts;
```

Comment:
Very nice idea to extract this logic in a variable 👍

Is there any plan to have a similar way to access API data as we have with Redux selectors? I can imagine we might want to use `isContributor` check in other places. At the moment we would have to duplicates what we have here. What I like the most about selectors is that they hide the structure of data representation. /cc @aduth

Reply:
Probably related to how with make `withApiData` state aware

Reply:
Selectors are just functions which happen to follow a particular first-argument convention. If it became a common pattern, I could see how it might be useful to create better organization around operating on API data. However, the implementation of `withAPIData` makes some compromises hinging on an assumption that its usage is very ad-hoc. To @youknowriad's point, if we were to make this more organized, we might similarly want to rethink how it works in the first place. One example might be to support a separate argument on the higher-order component akin to [`react-redux`'s `mapStateToProps`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) which helps provide a barrier between the component implementation and the underlying data structure.